File: uPark_client/entities/parking_slot.py
class ParkingSlot:
    def __init__(self, id = None, number = None, id_parking_lot = None, id_vehicle_type = None, reserved_disability = False):
        self._id = id
        self._number = number
        self._id_parking_lot = id_parking_lot
        self._id_vehicle_type = id_vehicle_type
        self._reserved_disability = reserved_disability


    def set_id(self, id):
        self._id = id

    # def set_number(self, number):
    #     self._number = number
    #
    # def set_id_parking_lot(self, id_parking_lot):
    #     self._id_parking_lot = id_parking_lot

    def set_id_vehicle_type(self, id_vehicle_type):
        self._id_vehicle_type = id_vehicle_type

    def set_reserved_disability(self, reserved_disability):
        self._reserved_disability = reserved_disability


    def get_id(self):
        return self._id

    def get_number(self):
        return self._number

    def get_id_parking_lot(self):
        return self._id_parking_lot

    def get_id_vehicle_type(self):
        return self._id_vehicle_type

    def get_reserved_disability(self):
        return self._reserved_disability

    def __eq__(self, other):
        if isinstance(other, ParkingSlot):
            return (self._id == other._id) or (self._number == other._number)
        return NotImplemented

    def __str__(self):
        parking_slot_info = (f"Parking slot info Id: {self._id}, number: {self._number}, id parking lot: {self._id_parking_lot}, id vehicle type: {self._id_vehicle_type}, "
        f"reserved disability: {self._reserved_disability}")
        return parking_slot_info

File: main_server/src/entities/parking_slot/parking_slot.cpp
#include "parking_slot.hpp"

ParkingSlot::ParkingSlot(){}
ParkingSlot::ParkingSlot(int id) : id(id) {}
ParkingSlot::ParkingSlot(int id, int number, int id_parking_lot, int id_vehicle_type, bool reserved_disability):
                                                                                id(id),
                                                                                number(number),
                                                                                id_parking_lot(id_parking_lot),
                                                                                id_vehicle_type(id_vehicle_type),
                                                                                reserved_disability(reserved_disability)
                                                                                {}

void ParkingSlot::setId(int id){
    this->id = id;
}

// void ParkingSlot::setNumber(int number){
//     this->number = number;
// }
//
// void ParkingSlot::setIdParkingLot(int id_parking_lot){
//     this->id_parking_lot = id_parking_lot;
// }

void ParkingSlot::setIdVehicleType(int id_vehicle_type){
    this->id_vehicle_type = id_vehicle_type;
}

void ParkingSlot::setReservedDisability(bool reserved_disability){
    this->reserved_disability = reserved_disability;
}


int ParkingSlot::getId() const{
    return id;
}

int ParkingSlot::getNumber() const{
    return number;
}

int ParkingSlot::getIdParkingLot() const{
    return id_parking_lot;
}

int ParkingSlot::getIdVehicleType() const{
    return id_vehicle_type;
}

bool ParkingSlot::getReservedDisability() const{
    return reserved_disability;
}


bool operator== (const ParkingSlot& ps1, const ParkingSlot& ps2)
{
    return ps1.id == ps2.id;
}

std::ostream& operator<<(std::ostream& os, const ParkingSlot& ps)
{
    os << "ParkingSlot - Id: " << ps.id << ", Number: " << ps.number << ", Id Parking lot: "
    << ps.id_parking_lot << ", Id Vehicle Type: " << ps.id_vehicle_type << ", Reserved disability: " << ps.reserved_disability << std::endl;
    return os;
}

// ParkingSlotException::ParkingSlotException(const std::string & message):_message(message) {}
// const char * ParkingSlotException::what() const throw() {
//       return _message.c_str();
// }
